python/src/nanoarrow/iterator.py

```diff
@@ -278,6 +398,50 @@ def _iter1(self, offset, length):
         return self._struct_tuple_iter(offset, length)
 
 
+def _get_tzinfo(tz_string, strategy=None):
+    # We can handle UTC without any imports
+    if tz_string.upper() == "UTC":
+        try:
+            # Added in Python 3.11
+            from datetime import UTC
+
+            return UTC
+        except ImportError:
+            from datetime import timedelta, timezone
+
+            return timezone(timedelta(0), "UTC")
+
+    # Try zoneinfo.ZoneInfo() (Python 3.9+)
+    if strategy is None or "zoneinfo" in strategy:
+        try:
+            from zoneinfo import ZoneInfo
+
+            return ZoneInfo(tz_string)
+        except ImportError:
+            pass
```

Comment:
If the timezone doesn't exist (Arrow C++ itself never does any validation of this, so this is typically "user input"), then this will actually raise a `ZoneInfoNotFoundError`, not `ImportError`. And you could also catch a KeyError to keep it generic.

```
In [76]: zoneinfo.ZoneInfo("bal")
...
ZoneInfoNotFoundError: 'No time zone found with key bal'

In [77]: zoneinfo.ZoneInfoNotFoundError.mro()
Out[77]: 
[zoneinfo._common.ZoneInfoNotFoundError,
 KeyError,
 LookupError,
 Exception,
 BaseException,
 object]
```

I suppose the ImportError was for older Python versions, but so I would do that (`import zoneinfo`) in a separate try/except

Reply:
Ah, but of course you are just catching the ImportError to then be able to try dateutil, and so the `ZoneInfoNotFoundError` will just be bubbled up to the user, which is fine. So ignore my comment ;)